include/s3select_oper.h

```diff
@@ -422,7 +422,7 @@ class value
     int64_t num;
     char* str;//TODO consider string_view
     double dbl;
-    boost::posix_time::ptime* timestamp;
+    std::tuple<boost::posix_time::ptime, boost::posix_time::time_duration, bool>* timestamp;
```

Comment:
its better to use typedef , it will ease readability and also future modifications

Reply:
Done